actors/builtin/power/power_actor.go

```diff
@@ -223,39 +145,39 @@ type DeleteMinerParams struct {
 }
 
 func (a Actor) DeleteMiner(rt Runtime, params *DeleteMinerParams) *adt.EmptyValue {
+
 	nominal, ok := rt.ResolveAddress(params.Miner)
 	if !ok {
 		rt.Abortf(exitcode.ErrIllegalArgument, "failed to resolve address %v", params.Miner)
 	}
 
-	var st State
-	rt.State().Readonly(&st)
+	ownerAddr, workerAddr := builtin.RequestMinerControlAddrs(rt, nominal)
+	rt.ValidateImmediateCallerIs(ownerAddr, workerAddr)
```

Comment:
Nooo, I don't think so. The miner _actor_ will be the only allowed caller here.

Reply:
I misunderstood this. Miner deletion is now a bit wierd: the caller has to call the power actor which then deletes the miner. I think it would be better to follow the same pattern as consensus fault reporting now does, where the caller invokes a method on the miner, and the miner notifies the power actor. But that's for later.